extensions/amp-experiment/1.0/mutation/attribute-mutation-default-url.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertAttributeMutationFormat} from './mutation';
+import {assertHttpsUrl} from '../../../../src/url';
+
+/**
+ * Mutation for attribute (url) mutations, for both
+ * the 'href' and 'src' attributes,
+ * on unspecifed elements.
+ *
+ * @implements {./mutation.Mutation}
+ */
+export class AttributeMutationDefaultUrl {
+  /**
+   * @param {!JsonObject} mutationRecord
+   * @param {!Array<Element>} elements
+   */
+  constructor(mutationRecord, elements) {
+    this.mutationRecord = mutationRecord;
```

Comment:
JSDoc please : )  And these can all be private variables

Reply:
So If I may ask, what are you looking for in the JSDoc? The structure of the mutation? 🤔 

And I made them public, because I access them for logging purposes for the user. But I can make them implement a toString() function 😄

Reply:
> So If I may ask, what are you looking for in the JSDoc? The structure of the mutation? 🤔

Maybe just `{!JsonObject}`. I'm asking it because I find we add JSDoc to all variables. I believe that's good to have to type check. 

> And I made them public, because I access them for logging purposes for the user. But I can make them implement a toString() function 😄
I found you convert to private variable. Thank you. But could you please point me to where `this.mutationRecord` is used outside the class? Thanks.